### PR TITLE
#24456 Handle adaption exceptions explicitly

### LINKED ISSUE #24456: handle exceptions from messageAdapter and ask adapter
Currently an exception from the adapter function is thrown out to the ActorAdapter before involving the Behavior. That means stopping. I think it's confusing that the Behavior's supervision strategy is not used for this case. It would be difficult to involve the Behavior, since this happens before.

I suggest that we catch, log and resume for such exceptions. One can think of the adapter as happening before receiving the actual message. One would have to do something terribly bad in the adapter function to corrupt the actor state so resume should be alright.

### PR DESCRIPTION
**Pull request by jrudolph**
Fixes #24456.

### CODE REVIEW

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala [hunk 1]
```diff
@@ -41,6 +47,7 @@ class AskSpec extends TestKit("AskSpec") with TypedAkkaSpec with ScalaFutures {
     "must fail the future if the actor is already terminated" in {
       val ref = spawn(behavior)
       (ref ? Stop).futureValue
+      TestProbe().expectTerminated(ref, 1.second)
```
Comment (patriknw) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala]: why 1.second, isn't there a default?
1 can be too small, and for no reason

**Review by patriknw — APPROVED** [review]
LGTM, test timeout nitpick

**Review by johanandren — APPROVED** [review]
LGTM after conflict resolved

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala (hunk 1 above)
Comment (jrudolph) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala]: Yeah, good point, didn't think about it.

**Pull request merged**